src/semgrep_agent/semgrep.py

```diff
@@ -78,15 +80,19 @@ def get_semgrepignore(ignore_patterns: List[str]) -> TextIO:
     return semgrepignore
 
 
-@dataclass
-class Results:
-    findings: FindingSets
-    total_time: float
+@attr.s(frozen=True)
+class Results(object):
+    findings = attr.ib(type=FindingSets)
+    total_time = attr.ib(type=float)
+    new = attr.ib(type=set, init=False)
+
+    def __attrs_post_init__(self) -> None:
+        object.__setattr__(self, "new", self.findings.expensive_new())
```

Comment:
Could we add a comment explaining why `self.new = self.findings.expensive_new()` wouldn't work? 😕